### PR TITLE
Added function dump(fstream*); added memset in reallocate();

### PR DESCRIPTION
**Pull request by Gguidini**
### Highlights
- Found bug when adding new bools that required bigger size of container. Fixed.
- Added dump() function to dump on file.

### Full Story
I came across your library when implementing Huffman's algorithm of compression for University Project. Really came in handy to write bit by bit arrays using the writeBool() function. 

I came across a bug, it had to do with memory garbage going into storage container when size was reallocated to a bigger size, because it copied everything from inside the container, but new spaces were left untouched. Because I was using things one bit at a time and the writeBool() uses bitwise or it created some bugs. I simply added a memset() to make sure new bits were all zero.

I also needed to write those arrays to a new file (the compressed file), so I added a new instance of the dump() function to write storage to file using fstream*.

Thanks you very much for this library, it was a real life saver. Very easy to use too, neat code.

### CODE REVIEW

File: compactstorage.cpp [hunk 1]
```diff
@@ -1,182 +1,204 @@
...
-	length = m_numBytes;
-
-	return result;
-}
-
-void CompactStorage::reset()
-{
-	m_curPos = 0;
-}
+/*
```
Comment (franzliedke) [inline compactstorage.cpp]: It looks like you (accidentally?) changed the line endings of this file. Please change back to CRLF, it makes the diff easier to understand.

**Review by franzliedke — CHANGES_REQUESTED** [review]
In addition, can I ask you to separate the bugfix and the new method into at least two commits?

File: compactstorage.cpp (hunk 1 above)
Comment (Gguidini) [reply compactstorage.cpp]: Yes, it was an accident, I'm sorry. Text editor issues. I'll fix it.

**Comment by Gguidini** [comment]
Rolled back changes and split them into 2 commits. Also got the line ending right.

**Review by franzliedke — APPROVED** [review]

**Comment by franzliedke** [comment]
Thank you!

Any pointers for me for how to make this library available through some sort of package manager? How is this usually done in C++ land?

**Comment by Gguidini** [comment]
I really don't know :grimacing: . I'm quite new in C++ land myself, sorry. But I quite enjoy header-only libraries such as yours. Especially for college - my case - I can see the source code and learn more, so I prefer it.

**Pull request merged**